File: doggis-ui/src/app/models/itemVenda.ts
import { Promocao, HistoricoPreco } from ".";
import { TipoItem } from './tipoItem';

export interface ItemVenda {
    id: number,
    descricao: string,
    foto: string,
    valor: number,
    promocao: Promocao,
    dataInclusao: Date,
    historico: HistoricoPreco[],
    tipo: TipoItem,
    totalEstoque?: number,
    patazBonus?: number,
}

File: doggis/src/main/java/br/pucminas/doggis/model/ItemVenda.java
package br.pucminas.doggis.model;

import java.util.Date;
import java.util.List;

public interface ItemVenda {
	public void setId(Long id);

	public Long getId();

	public void setDescricao(String descricao);

	public String getDescricao();
	
	public void setFoto(String foto);

	public String getFoto();

	public void setPromocao(Promocao promocao);

	public Promocao getPromocao();

	public void setValor(Double valor);

	public Double getValor();

	public Date getDataInclusao();

	public void setDataInclusao(Date dataInclusao);

	public List<HistoricoPreco> getHistorico();

	public void setHistorico(List<HistoricoPreco> historico);
	
	public TipoItem getTipo();
}
